sled-agent/src/common/instance.rs

```diff
@@ -280,20 +283,32 @@ impl InstanceStates {
                 // We don't return any action here for propolis to take
                 // because the actual migration process will be driven
                 // via a followup request to sled agent.
-                return Ok(None);
+                Ok(None)
             }
             InstanceState::Migrating => match self.current.migration_uuid {
-                // We're already performing the given migration: no-op
-                Some(id) if id == migration_id => return Ok(None),
-                // A different migration is already underway
-                Some(id) => {
-                    return Err(Error::InvalidRequest {
-                        message: format!(
-                            "migration already in progress: {}",
-                            id,
-                        ),
-                    });
+                Some(id) if id == migration_id => {
+                    match self.current.dst_propolis_uuid {
+                        // We're already performing the given migration to the
+                        // given propolis instance: no-op
+                        Some(id) if id == dst_propolis_id => Ok(None),
+                        // The migration ID matched but not the propolis ID: bail out
+                        Some(id) => Err(Error::InvalidRequest {
+                                message: format!(
+                                    "already perfoming given migration to different propolis: {}",
+                                    id
+                                )
+                            }),
+                        // If we're marked as 'Migrating' but have no dst propolis
+                        // ID bail because that shouldn't be possible
+                        None => Err(Error::internal_error(
+                            "migrating but no dst propolis id present"
+                        )),
+                    }
```

Comment:
Would love to start getting some tests for these conditions as soon as is reasonable

Reply:
Added.